### PR TITLE
chore: use cargo-binstall to install tools

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
to make it faster

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -501,16 +501,24 @@ script = """
 
 set -e
 
-for tool in cargo-llvm-cov cargo-nextest cargo-udeps cargo-hakari cargo-sort cargo-make cargo-upgrades typos-cli
+cargo install cargo-quickinstall
+cargo quickinstall cargo-binstall
```
Comment (xxchan) [inline Makefile.toml:505]: bootstrap as mentioned https://github.com/cargo-bins/cargo-binstall/issues/703

Comment (xxchan) [reply Makefile.toml:505]: `cargo-upgrades` is removed since I don't see it's used

File: Makefile.toml [hunk 2]
```diff
@@ -501,16 +501,24 @@ script = """
...
   echo "install: $(tput setaf 4)$tool$(tput sgr0)"
-  cargo install $tool --locked
+  cargo binstall -y --no-symlinks $tool
   echo
 done
 
-echo "install: $(tput setaf 4)risinglightdb/sqllogictest-rs$(tput sgr0)"
-cargo install --git https://github.com/risinglightdb/sqllogictest-rs --bin sqllogictest
-echo
+# Tools that fallback to `cargo install` when using `cargo binstall`.
```
Comment (ice1000) [inline Makefile.toml:514]: I think eventually they'll be installable with binstall.

Comment (xxchan) [reply Makefile.toml:514]: And `cargo binstall` doesn't use `--locked` when fallbacking to `cargo install`

Comment (ice1000) [reply Makefile.toml:514]: Ι see

**Review by TennyZhuang — APPROVED** [review]

**Pull request merged**